hack/vf-netns-switcher.sh

```diff
@@ -0,0 +1,343 @@
+declare -A pf_switch_ids
+
+TIMEOUT="${TIMEOUT:-2}"
+POLL_INTERVAL="1"
+
+while test $# -gt 0; do
+  case "$1" in
+
+   --netns | -n)
+      netnses+=("$2")
+      shift
+      shift
+      ;;
+
+   --pf | -d)
+      if [[ -z "${netnses[-1]}" ]];then
+          echo "ERROR: No netns is specified, please provide one using the --netns before using --pf."
+          exit 1
+      fi
+
+      pfs["${netnses[-1]}"]+="$2 "
+      shift
+      shift
+      ;;
+
+   --conf-file | -c)
+      conf_file=$2
+      shift
+      shift
+      ;;
+
+   --help | -h)
+      echo "
+vf-netns-switcher.sh --netns <> --pf <> [--conf-file <>]:
+
+	--netns | -n		Netns to switch the interface PFs and VFs to. This needs to be provided before the pfs
+				to associate the pfs to the netns. This flag can be repeated.
+
+	--pf | -d		The PF to switch it and its VFs to the specified netns. This must be provided after a
+				netns to associate the pf to the last provided netns. This option can be repeated.
+
+	--conf-file | -c	A file to read confs from, this will override cli flags. Conf file should be of the form:
+				- netns: <netns1>
+				  pfs:
+				  - <pf1>
+				  - <pf2>
+				- netns: <netns2>
+				  pfs:
+				  - <pf3>
+				  - <pf4>
+
+"
+      exit 0
+      ;;
+
+   *)
+      echo "No such option!!"
+      echo "Exitting ...."
+      exit 1
+  esac
+done
+
+get_pcis_from_pfs(){
+    local worker_netns="${1}"
+    shift
+    local interfaces="$@"
+    for interface in $interfaces; do
+        pcis["$interface"]="$(get_pci_from_net_name "$interface" "$worker_netns")"
+    done
+}
+
+get_pci_from_net_name(){
+    local interface_name=$1
+    local worker_netns="${2:-$netns}"
+
+    if [[ -z "$(ip l show $interface_name)" ]];then
+        if [[ -n "$(docker exec -t ${worker_netns} ip l show $interface_name)" ]];then
+            ip netns exec ${worker_netns} bash -c "basename \$(readlink /sys/class/net/${interface_name}/device)"
+            return 0
+        fi
+        echo ""
+        return 1
+    fi
+    basename $(readlink /sys/class/net/${interface_name}/device)
+}
+
+netns_create(){
+    local worker_netns="${1:-$netns}"
+
+    if [[ ! -e /var/run/netns/$worker_netns ]];then
+        local pid="$(docker inspect -f '{{.State.Pid}}' $worker_netns)"
+
+        if [[ -z "$pid" ]];then
+            return 1
+        fi
+
+        mkdir -p /var/run/netns/
+        rm -rf /var/run/netns/$worker_netns
+        ln -sf /proc/$pid/ns/net "/var/run/netns/$worker_netns"
+
+        if [[ -z "$(ip netns | grep $worker_netns)" ]];then
+            return 1
+        fi
+    fi
+    return 0
+}
+
+switch_pfs(){
+    local worker_netns="${1:-$netns}"
+    shift
+    local interfaces="$@"
+
+    for pf in $interfaces;do
+        switch_pf "$pf" "$worker_netns"
+    done
+}
+
+switch_pf(){
+    local pf_name="$1"
+    local worker_netns="${2:-$netns}"
+
+    if [[ -z "$(ip netns | grep ${worker_netns})" ]];then
+        echo "Namespace $worker_netns not found!"
```

Comment:
can you prefix all errors with either `Error:` or `ERROR:`  ? (pick one and stick with it :) )

Reply:
maybe add some helper funcs 

`log_error_exit()`
`log_error()`
`log()`

and use them throughout the code ?

Reply:
prefixed the error with Error, i dont think we will need to add a wrapper function, since as far as i can tell the file will not change alot.